services/QuillLMS/app/models/question_health_dashboard.rb

```diff
@@ -50,10 +38,8 @@ def cms_dashboard_stats
     @attempt_data ||= begin
```

Comment:
It looks like the intention here is to memoize the query result, but only the query itself is memoized

Reply:
Also, I'm not sure why the `begin` keyword is needed. You should be able to put the heredoc on the right side of the conditional assignment operator

Reply:
I misread the indentations. The memoization is correct, but it's hard to read.

Reply:
How about this?

```
  private def attempt_data
    query = <<-SQL etc etc SQL
    @attempt_data ||= ActiveRecord::Base.connection.execute(query).to_a

  end
```